src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -332,12 +337,14 @@ def preprocess(self, inputs, chunk_length_s=0, stride_length_s=None, ignore_warn
                 raise ValueError("Chunk length must be superior to stride length")
 
             # make sure that
-            for item in chunk_iter(inputs, self.feature_extractor, chunk_len, stride_left, stride_right):
+            for item in chunk_iter(inputs, self.feature_extractor, chunk_len, stride_left, stride_right, dtype):
                 yield item
         else:
             processed = self.feature_extractor(
                 inputs, sampling_rate=self.feature_extractor.sampling_rate, return_tensors="pt"
             )
+            if dtype is not None:
+                processed = {k: v.to(dtype=dtype) for k, v in processed.items()}
```

Comment:
Hi @Narsil,

This works fine for `seq2seq` models. 

But in case of `ctc` or `ctc_with_lm` model type, I think we should just cast `input_values` from `float32` to `float16`, and leave `attention_mask` to `int32` if it exists.

Reply:
Yes. And as above, if you directly use the `to` method on `processed`, it will take care of that for you.

Reply:
Done. Thanks, TIL